independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapMavenContext.java

```diff
@@ -143,7 +144,13 @@ public BootstrapMavenContext(BootstrapMavenContextConfig<?> config)
         this.remoteRepos = config.remoteRepos;
         this.remoteRepoManager = config.remoteRepoManager;
         this.cliOptions = config.cliOptions;
-        this.rootProjectDir = config.rootProjectDir;
+        if (config.rootProjectDir == null) {
+            final String topLevelBaseDirStr = PropertyUtils.getProperty(MAVEN_TOP_LEVEL_PROJECT_BASEDIR);
+            final Path tmp = topLevelBaseDirStr == null ? null : Paths.get(topLevelBaseDirStr);
+            this.rootProjectDir = tmp == null || !Files.exists(tmp) ? null : tmp;
```

Comment:
This is quite convoluted.

Shouldn't we just check if it's defined and if so throw an error if it doesn't exist?

Reply:
Makes sense. Done.